spec/javascript/components/create-nuage-cloud-subnet-form.test.jsx

```diff
@@ -0,0 +1,46 @@
+import CreateNuageCloudSubnetForm from '../../../app/javascript/components/create-nuage-cloud-subnet-form'
+import * as api from "../../../app/javascript/utils/api";
+
+const fetchRouterMock = jest.spyOn(api, 'fetchRouter').mockResolvedValue('SUCCESS');
+const dispatchMock = jest.spyOn(DEFAULT_STORE, 'dispatch');
+
+describe('CreateNuageCloudSubnetForm', () => {
+  describe('componentDidMount', () => {
+    it('router fetch succeeds', () => {
+      fetchRouterMock.mockResolvedValue({ ems_id: 111, ems_ref: 222 });
+      let component = renderComponent();
+      return fetchRouterMock().then(() => {
+        component.update();
+        expect(fetchRouterMock).toHaveBeenCalledWith(RECORD_ID);
```

Comment:
I am not against globals for testing, but i don't think its a good idea to store test specific in them values. It should be immediately clear what is inside such values. I think this value should be set in tests.